packages/react-devtools-shared/src/devtools/views/Profiler/ReloadAndProfileButton.js

```diff
@@ -43,8 +48,20 @@ export default function ReloadAndProfileButton() {
   const {
     recordChangeDescriptions,
     supportsReloadAndProfile,
+    supportsSynchronousXHR,
   } = useSubscription<SubscriptionData>(subscription);
 
+  const {dispatch: modalDialogDispatch} = useContext(ModalDialogContext);
+  useEffect(() => {
+    if (!supportsSynchronousXHR) {
+      modalDialogDispatch({
+        type: 'SHOW',
+        content:
+          'Synchronous XHR is required for reload and profile but is disabled on this site.',
+      });
+    }
+  }, [supportsSynchronousXHR, modalDialogDispatch]);
+
```

Comment:
Something feels off about the button looking okay until you click it, then showing a "nope doesn't work" message and disabling itself.

I think it would better for us to just disable the button with a different `title` instead.

```js
  const buttonTitle = supportsSynchronousXHR
    ? "Reload and start profiling"
    : "Reload and start profiling has been disabled because synchronous XHR is not supported on this site"
```

What are your thoughts?

Reply:
I agree it doesn't feel quite right. I originally tried as you suggest but the title doesn't show as a tooltip when the button is disabled.

Reply:
Hmm...Could have sworn that titles/tooltips show up for disabled buttons. Oooh, they only show up if you click the disabled button. Maybe that's not the worst thing. Hmm...I think a disabled button is probably still better than a temporarily enabled one like this.

Reply:
Makes sense - have updated it to always have the button disabled with the title you suggested above

Reply:
K thanks! I'll likely review this on Friday. Pretty busy this week but I'm trying to sit aside time on Fridays to catch up with DevTools stuff.